tests/test_nfft.py

```diff
@@ -2,53 +2,326 @@
 from unittest import TestCase, skipUnless
 from unittest.case import SkipTest
 
-from aspire.nfft import Plan, backend_available
+from aspire.nfft import backend_available
+from aspire.nfft import Plan
+
 
 import os.path
 DATA_DIR = os.path.join(os.path.dirname(__file__), 'saved_test_data')
 
 
 class SimTestCase(TestCase):
+
     def setUp(self):
-        pass
+        self.fourier_pts = np.array([
+            [ 0.88952655922411,  0.35922344760724, -0.17107966400962, -0.70138277562649],
+            [ 1.87089316522016,  1.99362869011803,  2.11636421501590,  2.23909973991377],
+            [-3.93035749861843, -3.36417300942290, -2.79798852022738, -2.23180403103185]
+        ])
+
+        self.vol = np.load(os.path.join(DATA_DIR, 'nfft_volume.npy'))
+
+        self.recip_space = np.array([-0.05646675 + 1.503746j, 1.677600 + 0.6610926j, 0.9124417 - 0.7394574j, -0.9136836 - 0.5491410j])
+
+        self.adjoint_vol = np.array([[[9.59495207e-01-1.06291322e+00j,  4.96001394e-01+1.60922425e+00j,
```

Comment:
This numpy array has more than 250 line. It might be better to use a small size or load it as a data file.

Reply:
Yes, I know, its not going to stay.  My opinion is that we shouldn't include bin files if not absolutely necessary.  A better test can be baked without any arrays stored; i just didn't have time to think about 3d.  It wasn't in the way of the hackathon.  In 1d you do something like the following up to scaling:

```
d = vol
dhat = fft(vol)
m = recip
mhat = ifft(recip)
<m,mhat> == <d,dhat>
```
You might figure this out for this case and add commit if you want. That would be helpful :)  @janden might know off his head....

Reply:
Previous we have many unit tests including npy data file. I believe we make sure that they are small. Some tests do need data in higher dimensions or many images. Good thing is that we have separated them in a subfolder and it is not necessary for standard delivery. Same things as the tutorials folder.

Reply:
That sounds good until you don't know why your repo is 5GB and you can't fix it :D. Whatever you guys like.  I think the mathematical test is more appropriate, but it can be done later.  Text diffs are easier to purge (space), I'd leave it, but Lets see what Joakim wants.  I sort of thought this was all too rough myself....otoh if using dev makes you guys more comfortable it is no difference to me. You can, for example, just branch off this branch... just saying.

Reply:
Yes. let's discuss it tomorrow.

Reply:
> Previous we have many unit tests including npy data file. I believe we make sure that they are small.

Right. This is better than including large binary files, of course, but I think it would be better to avoid hard-coded tests as much as possible. (This is what #156 is about for the block diagonal code.) One reason is size, yes (you're not getting those bytes in the repo back), but another reason is that hard-coded tests are not easy to inspect if something goes wrong. They don't check consistency, but to what extent the code matches with some fixed reference. This may be a more important issue than just size of the tests.